src/utils.py

```diff
@@ -80,3 +81,45 @@ def drop_na_samples(
     ).reset_index(drop=True)
 
     return profile
+
+
+def shuffle_features(feature_vals: np.array, seed: Optional[int] = 0) -> np.array:
+    """suffles all values within feature space
+
+    Parameters
+    ----------
+    feature_vals : np.array
+        shuffled
+
+    seed : Optional[int]
+        setting random seed
+
+    Returns
+    -------
+    np.array
+        Returns shuffled values within the feature space
+
+    Raises
+    ------
+    TypeError
+        Raised if a numpy array is not provided
+    """
+    # setting seed
+    np.random.seed(seed)
+
+    # shuffle given array
+    if not isinstance(feature_vals, np.ndarray):
+        raise TypeError("'feature_vals' must be a numpy array")
+    if feature_vals.ndim != 2:
+        raise TypeError("'feature_vals' must be a 2x2 matrix")
+
+    # creating a copy for feature vales to prevent overwriting of global variables
+    feature_vals = np.copy(feature_vals)
```

Comment:
Double checking, would it make sense to remove this copy due to `np.random.permutation` implicitly providing a new copy of the data ([link](https://stackoverflow.com/a/15474335))?

Reply:
Correct! Than you! Removed!